### ISSUE TITLE
Failed to activate the metrics package

### ISSUE THREAD

**Issue by Gruven** [atom/metrics#102]
*Edit by @rsese to add more specific repro steps and a link to @gmarton's notes*

### Steps to reproduce

1. Install atom-hg (apm install atom-hg)
2. Clone some Mercurial repository e.g. `hg clone https://selenic.com/repo/bloatwatch` (I installed [TortoiseHG](https://tortoisehg.bitbucket.io/) to do this on Windows)
3. Start Atom with that project or add it as a project folder

Reproduced on Windows 10 with Atom 1.32.1 and as noted by @gmarton, doesn't reproduce in 1.29.

See @gmarton's notes ("Took a quick look at both sources there seem to be a conflict between the atom-hg and mertics") at https://github.com/atom/metrics/issues/102#issuecomment-420023430.

---

[Enter steps to reproduce:]

1. Auto-update Atom from 1.29.0 to 1.30.0
2. Start Atom

**Atom**: 1.30.0 x64
**Electron**: 2.0.5
**OS**: Windows 7 SP1
**Thrown From**: [metrics](https://github.com/atom/metrics) package 1.6.1

### Stack Trace

Failed to activate the metrics package

```
At repository.getOriginURL is not a function

TypeError: repository.getOriginURL is not a function
    at subscriptions.add.atom.project.observeRepositories (~/AppData/Local/atom/app-1.30.0/resources/app/node_modules/metrics/lib/metrics.js:197:49)
    at Project.observeRepositories (~/AppData/Local/atom/app-1.30.0/resources/app/src/project.js:254:15)
    at Object.watchRepositories (~/AppData/Local/atom/app-1.30.0/resources/app/node_modules/metrics/lib/metrics.js:196:47)
    at Object.begin (~/AppData/Local/atom/app-1.30.0/resources/app/node_modules/metrics/lib/metrics.js:88:16)
    at ensureClientId (~/AppData/Local/atom/app-1.30.0/resources/app/node_modules/metrics/lib/metrics.js:29:42)
    at Object.ensureClientId (~/AppData/Local/atom/app-1.30.0/resources/app/node_modules/metrics/lib/metrics.js:106:13)
    at Object.activate (~/AppData/Local/atom/app-1.30.0/resources/app/node_modules/metrics/lib/metrics.js:29:16)
    at Package.activateNow (~/AppData/Local/atom/app-1.30.0/resources/app/src/package.js:230:33)
    at measure (~/AppData/Local/atom/app-1.30.0/resources/app/src/package.js:206:33)
    at Package.measure (~/AppData/Local/atom/app-1.30.0/resources/app/src/package.js:88:25)
    at activationPromise.Promise (~/AppData/Local/atom/app-1.30.0/resources/app/src/package.js:200:20)
    at new Promise (<anonymous>)
    at Package.activate (~/AppData/Local/atom/app-1.30.0/resources/app/src/package.js:198:38)
    at PackageManager.activatePackage (~/AppData/Local/atom/app-1.30.0/resources/app/src/package-manager.js:695:42)
    at config.transactAsync (~/AppData/Local/atom/app-1.30.0/resources/app/src/package-manager.js:670:36)
    at Config.transactAsync (~/AppData/Local/atom/app-1.30.0/resources/app/src/config.js:866:28)
    at PackageManager.activatePackages (~/AppData/Local/atom/app-1.30.0/resources/app/src/package-manager.js:668:23)
    at PackageManager.activate (~/AppData/Local/atom/app-1.30.0/resources/app/src/package-manager.js:647:50)
    at loadStatePromise.loadState.then (~/AppData/Local/atom/app-1.30.0/resources/app/src/atom-environment.js:875:27)
    at <anonymous>
```

### Commands

```
```

### Non-Core Packages

```
atom-hg 2.0.14 
city-lights-syntax 1.1.8 
city-lights-ui 1.5.3 
highlight-selected 0.14.0 
notepad-plus-plus-syntax 0.2.7 
seti-syntax 1.2.0 
seti-ui 1.11.0 
tortoise-hg 1.2.0 
```

**Comment by gmarton** [comment]
Removing atom-hg gets rid of the error for me. To test I started fresh with the atom-hg only to eliminate other conflicts.
**My Non-Core List:**
```
Community Packages (1)
└── atom-hg@2.0.14
```
Also disabled core module metrics but it did not help.

**Comment by rsese** [comment]
Thanks @gmarton - it looks like you and @Gruven have that package in common.

I [reset to factory defaults](https://flight-manual.atom.io/hacking-atom/sections/debugging/#reset-to-factory-defaults) and then installed `atom-hg` but I didn't see the error after starting Atom 1.30.0, I'm on macOS 10.12.6.  Do you have to be working with a Mercurial repository for the error to show?

**Comment by gmarton** [comment]
Yes, you have to be in a Mercurial repository, error does not show for me either otherwise.
I normally run the command atom . inside the repository on Debian Stretch.

**Comment by rsese** [comment]
Great, thanks for confirming @gmarton :v:

**Comment by scompo** [comment]
I can confirm the same behavior on windows 10.

Versions:
- Atom: 1.32.1 (x64)
- metrics: 1.6.2

**Comment by rsese** [comment]
Confirmed on Windows 10 with Atom 1.32.1 as well :+1:  Updated the issue body with more specific steps to reproduce and other details.

**Comment by s-m-e** [comment]
I can confirm this bug on Linux (openSUSE Leap 42.3 x86_64) with Atom 1.33.0. `atom-hg` and an actual Mercurial repository are required for triggering it.

**Comment by as-cii** [comment]
Thank you all for the report! 🙇

This was fixed via https://github.com/victor-torres/atom-hg/commit/6f32e504e851e0caadc3be6080530100258b4f57 and has been fixed since atom-hg@2.0.15.
